src/shortest_path.py

```diff
@@ -0,0 +1,233 @@
+"""Module with implementation of Weighted Graph."""
+from queue import Queue
+from stack import Stack
+from collections import OrderedDict
+import sys
+
+
+class Graph(object):
+    """Implementation of Graph Traversal."""
+
+    def __init__(self):
+        """."""
+        self.node_dict = {}
+
+    def nodes(self):
+        """Return a list of all nodes in the graph."""
+        return list(self.node_dict.keys())
+
+    def edges(self):
+        """Return a list of all edges in the graph."""
+        edge_list = []
+        for node1 in self.node_dict:
+            for node2 in self.node_dict[node1]:
+                edge_list.append((node1,
+                                  node2,
+                                  self.node_dict[node1][node2]))
+        return edge_list
+
+    def weight(self, n1, n2):
+        """."""
+        return self.node_dict[n1][n2]
+
+    def add_node(self, n):
+        """Add a node 'n' to the graph."""
+        self.node_dict.setdefault(n, OrderedDict())
+
+    def add_edge(self, n1, n2, weight=0):
+        """Add an edge to the graph with source, dest of 'n1', 'n2'. Add node if either not present."""
+        self.add_node(n1)
+        self.add_node(n2)
+        if n2 in self.node_dict[n1]:
+            raise ValueError("Edge already exists")
+        self.node_dict[n1][n2] = weight
+
+    def del_node(self, n):
+        """Delete the node 'n' from the graph. Raise error if no such node exists."""
+        if n in self.node_dict:
+            del self.node_dict[n]
+            for node in self.node_dict:
+                try:
+                    self.del_edge(node, n)
+                except:
```

Comment:
Is there a way to do this without using a Bare except? I like the catching of errors, but be specific when you do!